packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2095,6 +2109,11 @@ export default (superClass: Class<Parser>): Class<Parser> =>
           const typeArguments = this.tsParseTypeArguments();
 
           if (typeArguments) {
+            if (isOptionalCall && !this.match(tt.parenL)) {
+              error = [this.state.pos, tt.parenL];
```

Comment:
We only need to store the error position (in a `missingParenErrorPos` variable), and then later do 
```js
if (missingParenErrorPos) {
	this.unexpected(missingParenErrorPos, tt.parenL);
}
```